engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/stores/ExecutionStore.scala

```diff
@@ -156,8 +156,8 @@ sealed abstract class ExecutionStore private[stores](statusStore: Map[JobKey, Ex
   lazy val store: Map[ExecutionStatus, List[JobKey]] = statusStore.groupBy(_._2).safeMapValues(_.keys.toList)
   lazy val queuedJobsAboveThreshold = queuedJobs > MaxJobsToStartPerTick
 
-  def keysForNode(node: GraphNode): Iterable[JobKey] = {
-    statusStore.keys collect { case k if k.node eq node => k }
+  def backendJobDescriptorKeyForNode(node: GraphNode): Option[BackendJobDescriptorKey] = {
+    statusStore.keys collectFirst { case k: BackendJobDescriptorKey if k.node eq node => k }
```

Comment:
improvements: 

- stop once the first match is found
- don't collect up a bunch of results that are not `BackendJobDescriptorKey`s which would only get thrown away later but in the meantime need to be hashed for insertion into an associative collection
- single pass